src/compose/app.ts

```diff
@@ -546,47 +536,38 @@ export class App {
 				context.servicePairs,
 			);
 		} else {
-			if (!target) {
-				throw new InternalInconsistencyError(
-					'An empty changing pair passed to generateStepsForService',
-				);
-			}
-
+			// This service is in both current & target so requires an update,
+			// or it's a service that's not in target so requires removal
 			const needsSpecialKill = this.serviceHasNetworkOrVolume(
 				current,
 				context.networkPairs,
 				context.volumePairs,
 			);
-
 			if (
 				!needsSpecialKill &&
+				target != null &&
 				current.isEqualConfig(target, context.containerIds)
 			) {
 				// we're only starting/stopping a service
-				return this.generateContainerStep(current, target);
+				return this.generateContainerStep(current, target!);
 			}
 
-			let strategy =
-				checkString(target.config.labels['io.balena.update.strategy']) || '';
-			const validStrategies = [
-				'download-then-kill',
-				'kill-then-download',
-				'delete-then-download',
-				'hand-over',
-			];
-
-			if (!validStrategies.includes(strategy)) {
-				strategy = 'download-then-kill';
-			}
+			const strategy =
+				target == null
```

Comment:
This assignment of variables using ternary operators is pretty messy, can you think of a way of refactoring the code so it is more readable?